addons/mail/wizard/mail_template_preview.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class MailTemplatePreview(models.TransientModel):
+    _inherit = "mail.template"
+    _name = "mail.template.preview"
+    _description = "Email Template Preview"
+
+    @api.model
+    def _selection_target_model(self):
+        return [(model.model, model.name) for model in self.env['ir.model'].search([])]
+
+    @api.model
+    def _selection_languages(self):
+        return self.env['res.lang'].get_installed()
+
+    @api.model
+    def default_get(self, fields):
+        result = super(MailTemplatePreview, self).default_get(fields)
+        default_mail_template_id = self.env.context.get('default_mail_template_id')
+        if not default_mail_template_id:
+            return result
+        mail_template = self.env['mail.template'].browse(self.env.context['default_mail_template_id'])
+        res = self.env[mail_template.model_id.model].search([], limit=1)
+        if res and 'resource_ref' in fields:
```

Comment:
If `resource_ref` is not in `fields`, you do all that for nothing.

I would move it to the first condition:

```
if not default_mail_template_id or 'resource_ref' not in fields:
    return results
```